### DISCUSSION TITLE
Some sheet icons are not created properly

### DISCUSSION THREAD

**Discussion by mountaindude** [ptarmiganlabs/butler-sheet-icons#199] [Q&A]
## Question

After running Butler Sheet Icons (=BSI), *some* sheet icons look good (i.e. as miniatures of the actual sheet) but others don't look like the underlying sheet.

What's causing this?

## Answer

Look closer at those icons that weren't created properly.
Do they look like Sense sheets that are still loading?

If so you can try using a larger value in the `--pagewait` option. This value is in seconds.

This option controls how long BSI will wait after each interaction with the Sense app (this work is normally hidden to the user and happen behind the scenes every time BSI is executed).

For example, how BSI will wait after clicking the login button on the login page. 
Or how long to wait after navigating to a new sheet, before taking the screen shot that will later become a sheet icon.

So, if the chart objects on the sheet have not yet finished rendering when the screen shot is taken, you will get sheet icons that look like a still loading sheet.

Due to the nature of how Sense works it's difficult to determine when a chart or table is done rendering. 
It's thus difficult (or even impossible) for BSI to wait until the page is "ready". 
The best we can do is to set a large enough, fixed waiting value after which the screen shot is taken. 

This approach might seem like a waste of time as it really represents a worst case scenario that we need to handle.
Still, it usually doesn't matter if BSI uses 90 seconds or 30 seconds to process an app.

Bottom line is that you should rather wait a few extra seconds and get proper sheet icons than try to run BSI as quickly as possible, and risk poor sheet icons.
